src/fit.jl

```diff
@@ -92,15 +92,25 @@ function train!(
     start_iter = get_iter_number(estimator) + 1
     end_iter = start_iter + num_iterations - 1
 
-    for iter in start_iter:end_iter
+    metrics_idx_sequence = (((start_iter:end_iter) .- 1) .% estimator.metric_freq) .== 0
```

Comment:
You can do arithmetic on ranges without collection, i.e. you don't need to broadcast here until the equality.
```julia
metrics_idx_sequence = ((start_iter:end_iter - 1) % estimator.metric_freq) .== 0
```

Reply:
The example code you provided it subtracts 1 from end_iter as opposed to the whole range
When the range is put into parenthesis then nonbroadcast subtraction fails with no such method

```
julia> (1:5) -1
ERROR: MethodError: no method matching -(::UnitRange{Int64}, ::Int64)
Closest candidates are:
  -(::Complex{Bool}, ::Real) at complex.jl:307
  -(::Missing, ::Number) at missing.jl:115
  -(::Base.CoreLogging.LogLevel, ::Integer) at logging.jl:108
  ...
Stacktrace:
 [1] top-level scope at REPL[1]:1

julia> (1:5 -1)
1:4

julia> (1:5 -1) % 2
ERROR: MethodError: no method matching rem(::UnitRange{Int64}, ::Int64)
Closest candidates are:
  rem(::Any, ::Any, ::RoundingMode{:ToZero}) at div.jl:67
  rem(::Any, ::Any, ::RoundingMode{:Down}) at div.jl:68
  rem(::Any, ::Any, ::RoundingMode{:Up}) at div.jl:69
  ...
Stacktrace:
 [1] top-level scope at REPL[3]:1

julia> (1:5 -1) .% 2
4-element Array{Int64,1}:
 1
 0
 1
 0

julia> 
```